pkg/objstore/filesystem/filesystem.go

```diff
@@ -197,12 +196,15 @@ func (b *Bucket) Upload(_ context.Context, name string, r io.Reader) (err error)
 
 func isDirEmpty(name string) (ok bool, err error) {
 	f, err := os.Open(filepath.Clean(name))
-	if err != nil {
+	if os.IsNotExist(err) {
+		// The directory doesn't exist. We don't consider it an error and we treat it like empty.
+		return true, nil
+	} else if err != nil {
```

Comment:
Nit: since the first if returns, we normally style the second if without an else
```suggestion
	}
	if err != nil {
```

Reply:
Sure, addressed it!